server/src/main/java/org/opensearch/transport/InboundHandler.java

```diff
@@ -297,6 +300,23 @@ private static void sendErrorResponse(String actionName, TransportChannel transp
         try {
             response = handler.read(stream);
             response.remoteAddress(new TransportAddress(remoteAddress));
+
+            if (stream != EMPTY_STREAM_INPUT) {
```

Comment:
Treat unconsumed stream as (de)serialization failure and handle it as `TransportSerializationException`